cpp/src/lance/io/exec/take.cc

```diff
@@ -14,6 +14,10 @@
 
 #include "lance/io/exec/take.h"
 
+#include <arrow/result.h>
```

Comment:
Where r they used?

Reply:
In this same file, it uses arrow result multiple times.

Including the headers in `.cc` files can allow `.h` to just do forward declaration in general.